nova_vm/src/ecmascript/builtins/text_processing/string_objects/string_prototype.rs

```diff
@@ -877,16 +879,55 @@ impl StringPrototype {
         Ok(s.into_value())
     }
 
-    fn trim(_agent: &mut Agent, _this_value: Value, _: ArgumentsList) -> JsResult<Value> {
-        todo!()
+    /// ### [22.1.3.32 String.prototype.trim ( )](https://tc39.es/ecma262/#sec-string.prototype.trim)
+    fn trim(agent: &mut Agent, this_value: Value, _: ArgumentsList) -> JsResult<Value> {
+        // 1. Let S be the this value.
+        // 2. Return ? TrimString(S, start+end).
+        Self::trim_string(agent, this_value, TrimWhere::StartAndEnd)
     }
 
-    fn trim_end(_agent: &mut Agent, _this_value: Value, _: ArgumentsList) -> JsResult<Value> {
-        todo!()
+    /// #### [22.1.3.32.1 String.prototype.trimString ( )](https://tc39.es/ecma262/#sec-trimstring)
+    fn trim_string(agent: &mut Agent, value: Value, trim_where: TrimWhere) -> JsResult<Value> {
+        let s = {
+            // 1. Let str be ? RequireObjectCoercible(string).
```

Comment:
**suggestion:** I'd add a "fast path" check to see if there is anything to trim before doing `to_owned`. Often trim is called just to be sure, it'd be a shame to allocate in those cases.